### PR TITLE
docs: vstorage reference; chainStorage publishing guide

### LINKED ISSUE #947: Vstorage journey - 4daydapp
**User Story:** As a Developer I should be able to read the docs and understand what vstorage is, how I can leverage vstorage, how it should be used in the context of the dapp-agorics-basics

E.g. vstorage is a journey of requesting powers within a core-eval, creating a storage node, passing the storage node in privateArgs to a contract, using the storage node within a contract, reading from vstorage via raw HTTPS or ui-kit.

### PR DESCRIPTION
**Pull request by dckc**
closes #947

please excuse incremental work toward...

 - https://github.com/Agoric/documentation/issues/945

turning whitespace off in the diff should help.

subsumes:

 - #1015

### CODE REVIEW

**Comment by dckc** [comment]
## vstorage reference research notes

I'm about to delete https://github.com/Agoric/agoric-sdk/discussions/9115 , but some of it didn't make it into the PR, so let's keep it here...

@gibson042 wrote:

Yeah, I think you've pretty much got it, although it's worth noting that everything except `highPrioritySenders` (and maybe that as well) predates `published`, which I introduced to avoid collisions at the top level. Details that I know about:
* `activityhash` is a SwingSet concept; it is modified by [`emitCrankHashes`](https://github.com/Agoric/agoric-sdk/blob/1cd6408e27271160227f8aaa12d9771ea73a1da1/packages/swing-store/src/swingStore.js#L414-L443) after every delivery.
* For `highPrioritySenders`, the rubber meets the road in [`routeAction`](https://github.com/Agoric/agoric-sdk/blob/master/golang/cosmos/x/swingset/keeper/msg_server.go#L30-L41) (which pushes wallent spend actions from such addresses into a distinct comsic-swingset inbound queue that gets [drained before updating timers and processing non-priority actions](https://github.com/Agoric/agoric-sdk/blob/1cd6408e27271160227f8aaa12d9771ea73a1da1/packages/cosmic-swingset/src/launch-chain.js#L638-L669)).

I'm not aware of docs on any of them, nor even which are actively in use.

File: main/reference/vstorage-ref.md [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+      KREAdITEM: Object @Alleged: KREAdITEM brand#board00282 {},
+      USDC_axl: Object @Alleged: USDC_axl brand#board03040 {},
+      USDC_grv: Object @Alleged: USDC_grv brand#board04542 {},
+      USDT_axl: Object @Alleged: USDT_axl brand#board01744 {},
+      USDT_grv: Object @Alleged: USDT_grv brand#board03446 {},
+      stATOM: Object @Alleged: stATOM brand#board00990 {},
+      timer: Object @Alleged: timerBrand#board0425 {},
+    }
+
+> published.agoricNames.issuer has the well-known oracle brands.
```
Comment (Chris-Hibbert) [inline main/reference/vstorage-ref.md:155]:
```suggestion
> published.agoricNames.oracleBrand has the well-known oracle brands.
```

File: main/reference/vstorage-ref.md [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+    {
+      ATOM: Object @Alleged: Brand#board03935 {},
+      USD: Object @Alleged: Brand#board01034 {},
+      stATOM: Object @Alleged: Brand#board04388 {},
+    }
+
+> published.agoricNames.issuer shows denoms registered with the vbank.
```
Comment (Chris-Hibbert) [inline main/reference/vstorage-ref.md:163]:
```suggestion
> published.agoricNames.vbankAsset shows denoms registered with the vbank.
```

File: main/reference/vstorage-ref.md [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+        brand: Object @Alleged: IST brand#board0257 {},
+        value: 2750000n,
+      },
+      walletsProvisioned: 11n,
+    }
+
+## vstorage: wallet
+
+> The address of each provisioned smart wallet is a key under published.wallet.\*.
+> Here we show a hand-ful.
```
Comment (Chris-Hibbert) [inline main/reference/vstorage-ref.md:356]:
```suggestion
> Here we show a handful.
```

File: main/reference/zoe-api/zoe-helpers.md [hunk 4]
```diff
@@ -8,61 +8,66 @@ directly from **@agoric/zoe/src/contractSupport/index.js**. For example, the
...
-	 as long as either it gets the assets described in the want section of its proposal, or it retains
-     all of the assets specified in the give section of the proposal. This constraint applies to each
-     seat across the entire atomicRearrangement, not to the individual **TransferParts**.
-
-Note that you can construct the **TransferParts** that make up the *transfers* array manually, or for
+- All the mentioned seats are still live.
+- There aren't any outstanding stagings for any of the mentioned seats.
+
+      	Stagings are a reallocation mechanism that has been
+
```
Comment (Chris-Hibbert) [inline main/reference/zoe-api/zoe-helpers.md:33]: This inserted blank line breaks a paragraph. Please repair the paragraph.

File: main/reference/zoe-api/zoe-helpers.md [hunk 5]
```diff
@@ -378,3 +407,20 @@ const { userSeatPromise: AMMUserSeat, deposited } = zcf.offerTo(
   lenderSeat,
 );
 ```
+
+## prepareRecorderKitMakers(baggage, marshaller)
+
+Convenience wrapper for DurablePublishKit and Recorder kinds.
+
+NB: this defines two durable kinds. Must be called at most once per baggage.
```
Comment (Chris-Hibbert) [inline main/reference/zoe-api/zoe-helpers.md:415]:
```suggestion
::: tip Tip
This defines two durable kinds. Must be called at most once per baggage.
:::
```

File: main/reference/vstorage-ref.md [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+      provisionPool: 'provideds initial IST during smart wallet provisioning',
+      psm: 'see Inter Protocol',
+      reserve: 'see Inter Protocol',
+      vaultFactory: 'see Inter Protocol',
+      wallet: 'smart wallet status',
+    }
+
+## vstorage: agoricNames hubs
+
+> agoricNames contains several other NameHubs.
```
[the remark this answers is not in the document]
Comment (dckc) [reply main/reference/vstorage-ref.md:46]: they came from an ava snapshot test. but ok

File: main/reference/vstorage-ref.md [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+      'vbankAsset',
+    ]
+
+## vstorage: well known contracts
+
+> published.agoricNames.installation has the contents (entries)
+> of the Installation namehub. Here we show the object comprised
+> of those entries.
+> See also [agoricNames in vstorage](https://docs.agoric.com/guides/integration/name-services.html#agoricnames-in-vstorage)
+> regarding un-marshalling the data using board IDs.
```
[the remark this answers is not in the document]
Comment (dckc) [reply main/reference/vstorage-ref.md:64]: that edit seems to omit specifying that the contents are the entries of the namehub; i.e. an array of [key, value] pairs. hm.

[the remark this answers is not in the document]
Comment (dckc) [reply main/reference/vstorage-ref.md]: I word-smithed it a bit.

**Comment by Chris-Hibbert** [comment]
The preview seems broken. I find it easier to review with that open. Would you repair?

**Comment by dckc** [comment]
> The preview seems broken. 

oops! broken link.
it's rebuilding now.

File: main/guides/coreeval/permissions.md [hunk 8]
```diff
@@ -34,21 +34,21 @@ export const main = startSellConcertTicketsContract;
...
 
 - **board**: the [board](../integration/name-services.md#the-board-publishing-under-arbitrary-names) name service.
   **Note: the board only grows; no mechanism to reclaim storage has been established.**
 
-- **chainStorage**: to make storage nodes to write to vstorage.
+- **chainStorage**: to make storage nodes to [write to vstorage](../zoe/pub-to-storage).
   **Warning: this includes the right to over-write previously allocated storage nodes.**
 
-- **chainTimerService**: for getting the current timer and setting timer wake-ups; for example, at the conclusion of a governance vote.
+- **chainTimerService**: for getting the current [timer](../../reference/repl/timerServices) and setting timer wake-ups; for example, at the conclusion of a governance vote.
```
Comment (Chris-Hibbert) [inline main/guides/coreeval/permissions.md:45]: Aargh!  That's way out-of-date.  I added a link to the [internal doc](https://github.com/Agoric/agoric-sdk/blob/master/packages/SwingSet/docs/timer.md) to #948.

File: main/reference/vstorage-ref.md [hunk 9]
```diff
@@ -0,0 +1,434 @@
...
+    }
+```
+
+`published.agoricNames.oracleBrand` has the well-known oracle brands.
+
+    {
+      ATOM: Object @Alleged: Brand#board03935 {},
+      USD: Object @Alleged: Brand#board01034 {},
+      stATOM: Object @Alleged: Brand#board04388 {},
+    }
```
Comment (Chris-Hibbert) [inline main/reference/vstorage-ref.md:167]: missed the triple-quotes here.

File: main/reference/vstorage-ref.md [hunk 10]
```diff
@@ -0,0 +1,434 @@
...
+
+`published.agoricNames.vbankAsset` shows denoms registered with the vbank.
+
+```js
+    {
+      'ibc/295548A78785A1007F232DE286149A6FF512F180AF5657780FC89C009E2C348F': {
+        brand: Object @Alleged: USDC_axl brand#board03040 {},
+        denom: 'ibc/295548A78785A1007F232DE286149A6FF512F180AF5657780FC89C009E2C348F',
+        displayInfo: {
```
Comment (Chris-Hibbert) [inline main/reference/vstorage-ref.md:176]: you could show a subset without any loss.

File: main/reference/vstorage-ref.md [hunk 11]
```diff
@@ -0,0 +1,434 @@
...
+      },
+    }
+```
+
+## vstorage: provisionPool
+
+`published.provisionPool.governance` gives current values of governed params.
+See also Inter Protocol governance.
+
+    {
```
Comment (Chris-Hibbert) [inline main/reference/vstorage-ref.md:325]: triple-quotes here and for `.metrics`, please. I think it looks much better that way. Smaller, too!

File: main/reference/vstorage-ref.md [hunk 12]
```diff
@@ -0,0 +1,434 @@
...
+          decimalPlaces: 6,
+        },
+      },
+    }
+```
+
+## vstorage: provisionPool
+
+`published.provisionPool.governance` gives current values of governed params.
+See also Inter Protocol governance.
```
Comment (Chris-Hibbert) [inline main/reference/vstorage-ref.md:323]: missing link?

File: main/reference/vstorage-ref.md [hunk 13]
```diff
@@ -0,0 +1,434 @@
...
+              },
+            ],
+          },
+          brand: Object @Alleged: Zoe Invitation brand#board0074 {},
+        },
+      ],
+    }
+```
+
+The `published.wallet.${address}` key has wallet last update. For example:
```
Comment (Chris-Hibbert) [inline main/reference/vstorage-ref.md:411]: Is this accurate? Something was missing.

```suggestion
The `published.wallet.${address}` key has the wallet's last update. For example:
```

File: main/reference/zoe-api/zoe-helpers.md [hunk 14]
```diff
@@ -71,61 +74,63 @@ of the *transfer* argument of the **[atomicRearrange()](#atomicrearrange-zcf-tra
...
-    deprecated in favor of this **atomicRearrange()** function.
-    To prevent confusion, each reallocation can only be
-    expressed in the old way or the new way, but not a mixture.
- * Overall conservation must be maintained. In other words, the reallocated
-    **[Amounts](/reference/ertp-api/ertp-data-types#amount)** must balance out.
- * Offer Safety is preserved for each seat. That means reallocations can only take assets from a seat
-	 as long as either it gets the assets described in the want section of its proposal, or it retains
-     all of the assets specified in the give section of the proposal. This constraint applies to each
-     seat across the entire atomicRearrangement, not to the individual **TransferParts**.
+      	Stagings are a reallocation mechanism that has been
```
Comment (Chris-Hibbert) [inline main/reference/zoe-api/zoe-helpers.md:85]: This is still broken.  Compare:
stagings-wrong

== with ======

Stagings-orig

File: main/reference/vstorage-ref.md (hunk 9 above)
Comment (dckc) [reply main/reference/vstorage-ref.md:167]: good thing you're reading more carefully than I'm writing :)

File: main/guides/coreeval/permissions.md [hunk 15]
```diff
@@ -34,21 +34,21 @@ export const main = startSellConcertTicketsContract;
...
 In the top level promise space, we have:
 
-- **agoricNames**: read-only access to the [agoricNames](../integration/name-services.md#agoricnames-agoricnamesadmin-well-known-names) name service.
+- **agoricNames**: read-only access to the [agoricNames](../integration/name-services#agoricnames-agoricnamesadmin-well-known-names) name service.
 
 - **board**: the [board](../integration/name-services.md#the-board-publishing-under-arbitrary-names) name service.
   **Note: the board only grows; no mechanism to reclaim storage has been established.**
 
-- **chainStorage**: to make storage nodes to write to vstorage.
+- **chainStorage**: to make storage nodes to [write to vstorage](../zoe/pub-to-storage).
```
[the remark this answers is not in the document]
Comment (dckc) [reply main/guides/coreeval/permissions.md:42]: I suppose "right" suggests you're supposed to have it and "power" makes more sense even if you got it in ways that others didn't intend / expect. But I think "access" is more widely known than "power".

**Review by Chris-Hibbert — APPROVED** [review]

**Pull request merged**